Objects/exceptions.c

```diff
@@ -1864,6 +1864,41 @@ ImportError_reduce(PyObject *self, PyObject *Py_UNUSED(ignored))
     return res;
 }
 
+static PyObject *
+ImportError_repr(PyObject *self)
+{
+    int hasargs = PyTuple_GET_SIZE(((PyBaseExceptionObject *)self)->args) != 0;
+    PyObject *r = BaseException_repr(self);
+    PyImportErrorObject *exc = PyImportErrorObject_CAST(self);
+    PyUnicodeWriter *writer = PyUnicodeWriter_Create(0);
+    if (writer == NULL) goto error;
+    if (PyUnicodeWriter_WriteSubstring(writer, r, 0, PyUnicode_GET_LENGTH(r)-1) < 0) goto error;
+    if (exc->name) {
+        if (hasargs) {
+            if (PyUnicodeWriter_WriteASCII(writer, ", ", 2) < 0) goto error;
+        }
+        if (PyUnicodeWriter_WriteASCII(writer, "name='", 6) < 0) goto error;
+        if (PyUnicodeWriter_WriteSubstring(writer, exc->name, 0, PyUnicode_GET_LENGTH(exc->name)) < 0) goto error;
+        if (PyUnicodeWriter_WriteASCII(writer, "'", 1) < 0) goto error;
```

Comment:
For this one, as you're using the entire "exc->name", you can just use `PyUnicodeWriter_Format("name=%R", exc->name")` instead.